modules/data-table/test/spec.coffee

```diff
@@ -1858,99 +2111,564 @@ describe 'data-table', ->
+              cells: { 'name': 'Bob', 'age': 25, 'profession': 'Developer' }
+            }
+          ])
+          data.filteredCount.should.equal(3)
+          done()
+
+      it 'should get the rows from the data set with sorting and limited range', (done) ->
+        hx.dataTable.objectFeed(data).rows {start: 1, end: 2, sort: {column: 'name', direction: 'asc'}}, (data) ->
+          data.rows.should.eql([
+            {
+              id: 2,
+              cells: { 'name': 'Dan', 'age': 41, 'profession': 'Builder' }
+            },
+            {
+              id: 1,
+              cells: { 'name': 'Jan', 'age': 41, 'profession': 'Artist' }
+            }
+          ])
+          data.filteredCount.should.equal(3)
+          done()
+
+      it 'should get the rows from the data set with sorting and filtering', (done) ->
+        hx.dataTable.objectFeed(data).rows {start: 0, end: 2, sort: {column: 'name', direction: 'asc'}, filter: '41'}, (data) ->
+          data.rows.should.eql([
+            {
+              id: 2,
+              cells: { 'name': 'Dan', 'age': 41, 'profession': 'Builder' }
+            },
+            {
+              id: 1,
+              cells: { 'name': 'Jan', 'age': 41, 'profession': 'Artist' }
+            }
+          ])
+          data.filteredCount.should.equal(2)
+          done()
+
+      it 'should get the rows from the data set with sorting and filtering and limited range', (done) ->
+        hx.dataTable.objectFeed(data).rows {start: 0, end: 0, sort: {column: 'name', direction: 'asc'}, filter: '41'}, (data) ->
+          data.rows.should.eql([
+            {
+              id: 2,
+              cells: { 'name': 'Dan', 'age': 41, 'profession': 'Builder' }
+            }
+          ])
+          data.filteredCount.should.equal(2)
+          done()
+
+      it 'should get the rows for ids correctly 1', (done) ->
+        lookup = (row) -> row.id
+        hx.dataTable.objectFeed(data).rowsForIds [0, 1], lookup, (rows) ->
+          rows.should.eql([
+            {
+              id: 0, # hidden details can go here (not in the cells object)
+              cells: { 'name': 'Bob', 'age': 25, 'profession': 'Developer' }
+            },
+            {
+              id: 1,
+              cells: { 'name': 'Jan', 'age': 41, 'profession': 'Artist' }
+            }
+          ])
+          done()
+
+      it 'should get the rows for ids correctly 2', (done) ->
+        lookup = (row) -> row.id
+        hx.dataTable.objectFeed(data).rowsForIds [2, 0], lookup, (rows) ->
+          rows.should.eql([
+            {
+              id: 2,
+              cells: { 'name': 'Dan', 'age': 41, 'profession': 'Builder' }
+            },
+            {
+              id: 0, # hidden details can go here (not in the cells object)
+              cells: { 'name': 'Bob', 'age': 25, 'profession': 'Developer' }
+            }
+          ])
+          done()
+
+      it 'should use the provided filter function', ->
```

Comment:
New tests for object feeds start here - the ones above haven't changed but have been moved in the file